common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/CurrentTaskAuthorizationActivity.java

```diff
@@ -0,0 +1,163 @@
+
+    private CurrentTaskBrowserAuthorizationFragment mFragment;
+    private boolean mCloseCustomTabs = true;
+    private BroadcastReceiver redirectReceiver;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        final Fragment fragment = AuthorizationActivityFactory.getAuthorizationFragmentFromStartIntent(getIntent());
+
+        if(fragment instanceof CurrentTaskBrowserAuthorizationFragment){
+            mFragment = (CurrentTaskBrowserAuthorizationFragment) fragment;
+            mFragment.setInstanceState(getIntent().getExtras());
+        }else{
+            IllegalStateException ex = new IllegalStateException("Unexpected fragment type");
+            Logger.error(TAG, "Fragment provided was not of type CurrentTaskBrowserAuthorizationFragment", ex);
+            throw ex;
+        }
+
+        //This will be invoked when CurrentTaskAuthorizationActivity is not found in the current task stack... hence it will create fresh
+        if (REDIRECT_RETURNED_ACTION.equals(getIntent().getAction())) {
+            if(CurrentTaskBrowserAuthorizationFragment.class.isInstance(mFragment)) {
+                Bundle arguments = new Bundle();
+                arguments.putBoolean("RESPONSE", true);
+                mFragment.setArguments(arguments);
+                mFragment.completeAuthorizationInBrowserFlow(getIntent().getStringExtra("RESPONSE_URI"));
+                finish();
+                return;
+            }
+
+            return;
+        }
+
+        setFragment(mFragment);
+
+        if (savedInstanceState == null) {
+
+            mCloseCustomTabs = false;
+
+            // This activity will receive a broadcast if it can't be opened from the back stack
+            redirectReceiver = new BroadcastReceiver() {
+                @Override
+                public void onReceive(Context context, Intent intent) {
+                    // Remove the custom tab on top of this activity.
+                    Intent newIntent = new Intent(CurrentTaskAuthorizationActivity.this, CurrentTaskAuthorizationActivity.class);
+                    newIntent.setAction(REFRESH_TO_CLOSE);
+                    newIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+                    startActivity(newIntent);
+                }
+            };
+            LocalBroadcastManager.getInstance(this).registerReceiver(redirectReceiver,
+                    new IntentFilter(REDIRECT_RETURNED_ACTION)
+            );
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (!mFragment.onBackPressed()) {
+            super.onBackPressed();
+        }
+    }
+
+    /**
+     * This is invoked when an existing activities is re-used and provided with a new intent with additinoal information
+     * NOTE: It's important that you use setIntent update the intent associated with the activity.  Otherwise subsequent calls to
```

Comment:
```suggestion
     * NOTE: It's important that you use setIntent to update the intent associated with the activity.  Otherwise subsequent calls to
```